File: java/src/main/java/abstract_factory/product/SuperTruck.java
package abstract_factory.product;

import abstract_factory.product.part.VehiclePartFactory;

import java.util.List;

public class SuperTruck extends Vehicle {
    public static final String NAME = "Super Truck";
    private final VehiclePartFactory vehiclePartFactory;

    public SuperTruck(VehiclePartFactory vehiclePartFactory) {
        name = NAME;
        this.vehiclePartFactory = vehiclePartFactory;
    }

    @Override
    public void assembly() {
        engine = vehiclePartFactory.createEngine();
        tire = vehiclePartFactory.createTire();
        options = List.of();
    }
}

File: src/main/kotlin/abstract_factory/product/SuperTruck.kt
package abstract_factory.product

import abstract_factory.product.part.VehiclePartFactory

class SuperTruck(private val vehiclePartFactory: VehiclePartFactory) : Vehicle() {
    override var name = NAME

    override fun assembly() {
        engine = vehiclePartFactory.createEngine()
        tire = vehiclePartFactory.createTire()
        options = listOf()
    }

    companion object {
        const val NAME: String = "Super Truck"
    }
}